x-pack/plugins/secops/server/graphql/kpi_network/resolvers.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { GraphQLResolveInfo } from 'graphql';
+import { omit } from 'lodash/fp';
+
+import { Source } from '../../graphql/types';
+import { FrameworkRequest, internalFrameworkRequest } from '../../lib/framework';
+import { KpiNetwork } from '../../lib/kpi_network';
+import { KpiNetworkAdapter } from '../../lib/kpi_network/types';
+import { SourceStatus } from '../../lib/source_status';
+import { Sources } from '../../lib/sources';
+import { createSourcesResolvers } from '../sources';
+import { SourcesResolversDeps } from '../sources/resolvers';
+import { mockSourcesAdapter, mockSourceStatusAdapter } from '../sources/resolvers.test';
+import { mockKpiNetworkData, mockKpiNetworkFields } from './kpi_network.mock';
+import { createKpiNetworkResolvers, KpiNetworkResolversDeps } from './resolvers';
+
+const mockGetFields = jest.fn();
+mockGetFields.mockResolvedValue({ fieldNodes: [mockKpiNetworkFields] });
+jest.mock('../../utils/build_query/fields', () => ({
+  getFields: mockGetFields,
+}));
+
+const mockGetKpiNetwork = jest.fn();
+mockGetKpiNetwork.mockResolvedValue({
+  KpiNetwork: {
+    ...mockKpiNetworkData.KpiNetwork,
+  },
+});
+const mockKpiNetworkAdapter: KpiNetworkAdapter = {
+  getKpiNetwork: mockGetKpiNetwork,
+};
+
+const mockKpiNetworkLibs: KpiNetworkResolversDeps = {
+  kpiNetwork: new KpiNetwork(mockKpiNetworkAdapter),
+};
+
+const mockSrcLibs: SourcesResolversDeps = {
+  sources: new Sources(mockSourcesAdapter),
+  sourceStatus: new SourceStatus(mockSourceStatusAdapter, new Sources(mockSourcesAdapter)),
+};
+
+const req: FrameworkRequest = {
+  [internalFrameworkRequest]: {
+    params: {},
+    query: {},
+    payload: {
+      operationName: 'test',
+    },
+  },
+  params: {},
+  query: {},
+  payload: {
+    operationName: 'test',
+  },
+};
+
+const context = { req };
+
+describe('Test Source Resolvers', () => {
+  test('Make sure that getKpiNetwork have been called', async () => {
+    const source = await createSourcesResolvers(mockSrcLibs).Query.source(
+      {},
+      { id: 'default' },
+      context,
+      {} as GraphQLResolveInfo
+    );
+    const data = await createKpiNetworkResolvers(mockKpiNetworkLibs).Source.KpiNetwork(
+      source as Source,
+      {
+        timerange: {
+          interval: '12h',
+          to: 1514782800000,
+          from: 1546318799999,
+        },
+      },
+      context,
+      {} as GraphQLResolveInfo
+    );
+    expect(mockKpiNetworkAdapter.getKpiNetwork).toHaveBeenCalled();
+    expect(data).toEqual(omit('status', mockKpiNetworkData));
+  });
+});
```

Comment:
Awesome tests!

I can give some context around things we've learned along our journey for context. No changes requested since you're following what we're already doing (consistency > correctness in most cases), but context is usually helpful to see where we're heading and better ways to do things and you might have better ways of doing things than what we're doing at the moment so we can sweep through and changes things.

The `omit` is still ok to have but we found out it looks like it will eventually be removed from lodash and we will have to remove it from our code base:
https://github.com/lodash/lodash/issues/2930

I removed some from my pull requests in favor of `.cloneDeep` and `delete`. It doesn't make me feel good as it feels less functional and react/redux style but there's this git page:
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_omit

which shows some destructuring tricks we might be able to seep into our code a touch to reduce usage. I haven't used it yet but maybe that's better?

The other context data point is that In some of our tests and areas of our code we are using the `as casting` such as `as GraphQLResolveInfo` but we have to be careful about them as they usually show us that we have a mistake somewhere in our typings. They're not forbidden but we are being cautious about their usage. 

This outlines the caveats with `as casting` I have read that made me more careful:
https://basarat.gitbooks.io/typescript/docs/types/type-assertion.html

In a lot of cases we might be able to use a `Partial` or other tricks?
https://www.typescriptlang.org/docs/handbook/advanced-types.html

Overall in tests we are not going to avoid the `as` casts as we do interesting things with the data. I am just pointing out context for knowledge...

Reply:
Thanks for this @FrankHassanabad! 
For the first thing with `omit`, I'm thinking of replacing it with something like this:
```
describe('Test Source Resolvers', () => {
  test('Make sure that getKpiNetwork have been called', async () => {
    const source = await createSourcesResolvers(mockSrcLibs).Query.source(
      {},
      { id: 'default' },
      context,
      {} as GraphQLResolveInfo
    );
    const { status, ...expectedKpiNetworkData } = mockKpiNetworkData;
    ....    

    expect(data).toEqual(expectedKpiNetworkData);
  });
});
```
However ends up it complains about `status` is not even exists.
I'm a bit confused as this case seems to be part of the existing unit tests, but I couldn't find the `status` key from their mock data either. Should I just update the mock data with this key or actually I don't need this case?

Reply:
It looks like the types are telling the truth. In all of the 5 places we are using it, it's not needed. I think that we might have had status at one point but not anymore. Let's just remove the omit statement altogether and everything should be great.

Nice to see that using typings shows more left over code we don't need! :-)

Reply:
Alrighty @franekrichardson , I'll check the mock data from our existing cases and update them as well.

Reply:
Haha. I think that was a different Frank you pinged up there.

Reply:
Blimey! Haha sorry about that lol

Reply:
not just a frank but a franek no less :-)

anyway I'll just drive by hi :wave: and leave ya'll to it :bowing_man:

Reply:
👋 👋 👋 😄